cpp/tests/streams/io/orc_test.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/io/detail/orc.hpp>
+#include <cudf/io/orc.hpp>
+#include <cudf/io/orc_metadata.hpp>
+#include <cudf/io/orc_types.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/default_stream.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <iostream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <vector>
+
+auto const temp_env = static_cast<cudf::test::TempDirTestEnvironment*>(
+  ::testing::AddGlobalTestEnvironment(new cudf::test::TempDirTestEnvironment));
+
+class ORCTest : public cudf::test::BaseFixture {};
+
+template <typename T>
+inline auto random_values(size_t size)
```

Comment:
Generating random values seems overkill for stream testing since the values themselves will not influence how the stream is passed through and the results are not checked.
Perhaps just hardcode a set of values instead?

Reply:
Even with random generated input, this is not what we are doing. We already have `generate_input_tables` for on-GPU random data generation.

Reply:
+1
Assuming that the goal is to call all kernels that ORC reader/writer ever call, we can just pass all zeroes. 
Another benefit from non-random values: strings column with all equal, non-empty strings would exercise the dictionary encode/decode path, increasing the test code coverage.

Reply:
Thank you for the feedback! I've modified the tests to take in constant columns, and that gave rise to stream pool errors in the dictionary encoder calls. Missed this case in the previous tests!